### ISSUE TITLE
Set different ecmaversion for files

### ISSUE THREAD

**Issue by Eschon** [eslint/eslint#5651]
I'm currently using eslint 2.4.0 and most of my files are written in es6 so I've set the ecmaversion to 6 in my config, but I also have a few files that are es5 and they show a lot of errors because of that.

I searched around for a bit but as far as I know there is no way to set the ecmaversion to 5 for these files.
Would it be possible to implement something like this?

```
/* ecmaversion 5 */

... es5 code ...
```

**Comment by mysticatea** [comment]
Thank you for this issue.

Unfortunately, the way to overwrite parser options from the inside of files does not exist for now.
Maybe we can revisit #3698.
Or if the es5 files are collected in a directory, we can put `.eslintrc` into the directory.

By the way, es6 has high compatibility to es5.
What errors did you get?
If the errors are reported by rules for es6, parser options don't turn the rules off, so you still have to turn the rules off manually. e.g. `/*eslint no-var:0*/`.

**Comment by Eschon** [comment]
Thank you for your fast answer.

Since most of the es5 files are in a directory without any es6 files an extra `.eslintrc` file would be a viable option. 

Like you already guessed most errors are `no-var` and `prefer-arrow-callback`. Turning those of manually in the files would work too.

I think your comment solves my issues. I'll keep the issue open in case you think this feature would still be useful and should be implemented. Otherwise feel free to close it.

**Comment by mysticatea** [comment]
#3698 is still opened, so I close this issue.
Thank you!

**Comment by barbalex** [comment]
I am having the same issue because I am using CouchDB and CouchDB views and lists need to be written in es5.

As I have moved all views and lists into separate directories I tried adding a `eslint.json`-File to these directories declaring ecma 5 like this:

in top project folder:

    {
      "extends": "airbnb",
      "env": {
        "node": true
      },
      "rules": {
        "react/jsx-no-bind": 0,
        "comma-dangle": 0,
        "no-use-before-define": 0,
        "consistent-return": 0,
        "semi": [2, "never"],
        "quotes": [2, "single"],
        "strict": 0,
        "max-len": ["error", 120, 2, {"ignoreUrls": true}],
        "no-underscore-dangle": ["error", { "allow": ["_id", "_rev", "_attachments"] }],
        "no-param-reassign": ["error", { "props": false }]
      }
    }

in views folder:

    {
      "parserOptions": {
        "ecmaVersion": 5
      }
    }

Unfortunately I am still getting `no-var` and `func-names` errors. 
What am I doing wrong?

**Comment by platinumazure** [comment]
@barbalex Please open a new issue and follow the issue template.
